test/ng/qSpec.js

```diff
@@ -698,6 +753,18 @@ describe('q', function() {
 
                 expect(logStr()).toBe('finally1()->{}; success2(foo)->foo');
               });
+
+              it("should untrack the promise", function() {
+                var promise = createPromise();
+                var promise2 = createPromise();
+                var untrackBefore = mockPromiseTracker.untrackedPromises.length;
+                resolve2('bar');
+                 promise['finally'](fin(1, promise))
+                       .then(success(2));
+                 resolve('foo');
+                mockNextTick.flush();
+                 expect(mockPromiseTracker.untrackedPromises.length).toBeGreaterThan(untrackBefore);
```

Comment:
I'm not really a fan of this, can we be explicit in what the length should be? It should be exact. Unit testing like this may lead to `1=1` type tests